src/dynamic_loading_c_impl.rs

```diff
@@ -159,6 +159,9 @@ impl<T> CKBDLContext<T> {
     }
 
     pub fn load<'a>(&'a mut self, dep_cell_data_hash: &[u8]) -> Result<Library, Error> {
-        self.load_with_offset(dep_cell_data_hash, 0, size_of::<CKBDLContext<T>>())
+        self.load_with_offset(dep_cell_data_hash, 0, size_of::<CKBDLContext<T>>(), 0)
```

Comment:
Personally, I wasn't sure about the API design here.

* First of all things, `self.load_with_offset(dep_cell_data_hash, 0, size_of::<CKBDLContext<T>>(), 0)` seems to only deal with the hash type of `data` here. With Mirana hardfork we have `data1`, in the future there might be more `data` versions. Hardcoding `0` here seems quite misleading.
* If you think about it, the correct way to locate a code cell unambiguously, is really using a combination of `code_hash` and `hash_type`, requiring only data hash or type id, seems quite confusing to me, in the future there might also be the need for more functions here.

So my personal suggestion here, is that we should deprecate `load` here, since it only cope with `data` as a hash type. After that, we should add an unambiguous load function that takes `code_hash` and `hash_type` as arguments. There might be wrappers like `load_by_type_id` on top of that, but having a load function taking both `code_hash` and `hash_type`, IMO, is a better choice.

Reply:
According to: https://github.com/nervosnetwork/ckb-lua/blob/b71d44288e8fb085074d3af056b359b8c6c22afa/include/ckb-c-stdlib/ckb_syscalls.h#L189-L190

The meaning of last argument is "to fetch data through type id or not". It's not same as `hash_type`.

Reply:
I'm afraid I'm not following you here.

https://github.com/nervosnetwork/ckb-std/blob/64236de2121f295de1bd129b7477611ffcbb4b74/src/dynamic_loading_c_impl.rs#L121-L127
https://github.com/nervosnetwork/ckb-lua/blob/b71d44288e8fb085074d3af056b359b8c6c22afa/include/ckb-c-stdlib/ckb_syscalls.h#L189-L190

In both cases, the name of the argument is clearly `hash_type`. I can also treat the C code as parsing `hash_type` exactly, both interpretation works.

Reply:
It should be renamed to "from_type_id".

Reply:
I think the real question we should ask is: should we pass in `hash_type` or a boolean variable? I would personally think that using `hash_type` is a better and future-proof way.